solr/docker/tests/shared.sh

```diff
@@ -90,18 +90,44 @@ function prepare_dir_to_mount {
   fi
   rm -fr "$folder" >/dev/null 2>&1
   mkdir "$folder"
-  #echo "***** Created varsolr folder $PWD / $folder"
+  #echo "***** Created varsolr folder $BUILD_DIR / $folder"
 
   # The /var/solr mountpoint is owned by solr, so when we bind mount there our directory,
   # owned by the current user in the host, will show as owned by solr, and our attempts
   # to write to it as the user will fail. To deal with that, set the ACL to allow that.
   # If you can't use setfacl (eg on macOS), you'll have to chown the directory to 8983, or apply world
   # write permissions.
-  if [[ "$OSTYPE" == "darwin"* ]]; then
-    # Workaround for macOS
-    sudo chmod -R 777 "$folder"
-    sudo chown -R "$userid" "$folder"
-  else
+  if command -v setfacl &> /dev/null; then
     setfacl -m "u:$userid:rwx" "$folder"
   fi
 }
+
+
+# Shared setup
```

Comment:
Are you making general improvements to the test scripts, or do they relate to necessities of the port to lucene-solr?  I suggest suppressing your natural urge to improve code in a porting job to limit to just do the port.  Future work items can refactor common code out of bash scripts or whatever else.

Reply:
These changes were necessary for separating out the test output from the source directory from the tests as well as having unique container names for the tests. This was needed in the migration, because the tests assumed that you were running in a particular directory layout before (the layout in docker-solr). Moving to gradle, this needed to be a lot more flexible.

Reply:
But I am happy to leave the testing stuff to a different PR if that will be easier to review.

Reply:
Your justification makes sense.  The cause of those changes weren't clear to me before.